packages/cardpay-reward-programs/cardpay_reward_programs/rule.py

```diff
@@ -0,0 +1,90 @@
+import hashlib
+import json
+from abc import ABC, abstractmethod
+
+import duckdb
+import pandas as pd
+from cardpay_reward_programs.utils import get_files
+
+
+class Rule(ABC):
+    """
+    A single image should run only a single rule
+    """
+
+    def __init__(self, core_parameters, user_defined_parameters):
+        self.set_core_parmaters(**core_parameters)
+        self.set_user_defined_parameters(**user_defined_parameters)
+
+    def set_core_parmaters(
+        self, subgraph_config_location, payment_cycle_length, valid_from, valid_to, token
+    ):
+        self.subgraph_config_location = subgraph_config_location
+        self.payment_cycle_length = payment_cycle_length
+        self.valid_from = valid_from
+        self.valid_to = valid_to
+        self.token = token
```

Comment:
I don't think token should be part of the core params, that would be down to the rule which token(s) it should use.

Reply:
ok sure i'll change it!